src/main/scala/org/constellation/primitives/EdgeDAO.scala

```diff
@@ -1,15 +1,178 @@
 package org.constellation.primitives
 
-import java.util.concurrent.Executors
+import java.util.concurrent.{Executors, TimeUnit}
 
-import org.constellation.ProcessingConfig
+import akka.util.Timeout
+import org.constellation.consensus.EdgeProcessor.acceptCheckpoint
+import org.constellation.consensus.{Snapshot, SnapshotInfo, TipData}
 import org.constellation.primitives.Schema._
+import org.constellation.{DAO, ProcessingConfig}
 
 import scala.collection.concurrent.TrieMap
 import scala.concurrent.{ExecutionContext, ExecutionContextExecutor}
+import scala.util.Random
+
+
+class ThreadSafeTXMemPool() {
+
+  private var transactions = Seq[Transaction]()
+
+  def pull(minCount: Int): Option[Seq[Transaction]] = this.synchronized{
+    if (transactions.size > minCount) {
```

Comment:
>= ?

Reply:
Wasn't sure how this would affect splitAt but maybe it's fine to change